integration-tests/utils/integration-helpers.js

```diff
@@ -133,22 +133,62 @@ const deleteWorkspace = withSignedInPage(async ({ page, billingProject, workspac
   }
 });
 
-const deleteWorkspaceV2 = withSignedInPage(async ({ page, billingProject, workspaceName }) => {
+/* TODO
+const deleteWorkspaceInUi = async ({ page, testUrl, token, workspaceName }) => {
+  await gotoPage(page, testUrl);
+  await findText(page, 'View Workspaces');
+  await viewWorkspaceDashboard(page, token, workspaceName);
+  await findText(page, clickable({ textContains: 'Delete' }));
+  await noSpinnersAfter(page, {
+    action: () => click(page, clickable({ textContains: 'Delete' })),
+    timeout: Millis.ofMinute,
+    debugMessage: 'deleteWorkspaceInUi - opened delete',
+  });
+  await findText(page, "Please type 'Delete Workspace' to continue:");
+  await fillIn(page, input({ placeholder: 'Delete Workspace' }), 'Delete Workspace');
+  await noSpinnersAfter(page, {
+    action: () => click(page, clickable({ textContains: 'Delete Workspace' })),
+    timeout: Millis.ofMinutes(2),
+    debugMessage: 'deleteWorkspaceInUi - submitted delete',
+  });
+};
+*/
+
+const deleteWorkspaceV2AsUser = async ({ page, billingProject, workspaceName }) => {
   try {
-    await page.evaluate(
-      async (name, billingProject) => {
-        await window.Ajax().Workspaces.workspaceV2(billingProject, name).delete();
-      },
-      workspaceName,
-      billingProject
-    );
-    console.info(`Deleted workspace: ${workspaceName}`);
+    const isAppsEmpty = await deleteAppsV2({ page, billingProject, workspaceName });
+    const isRuntimesEmpty = await deleteRuntimesV2({ page, billingProject, workspaceName });
+    if (isAppsEmpty && isRuntimesEmpty) {
+      // TODO [IA-4337] fix disk delete as part of runtime delete
+      // Once that's done, should not need to wait
+      await delay(Millis.ofMinutes(3));
+      await page.evaluate(
+        async (workspaceName, billingProject) => {
+          try {
+            await window.Ajax().Workspaces.workspaceV2(billingProject, workspaceName).delete();
+          } catch (error) {
+            if (error.status === 404) {
```

Comment:
I chose to interpret `404` as indicating a successful deletion. However, I'm seeing that 404 responses are also returned when the workspace exists but Terra shows that Google permissions are syncing. Should my test not make this assumption?
![Screenshot 2023-11-28 at 4 09 10 PM](https://github.com/DataBiosphere/terra-ui/assets/117680805/bc518b29-28b8-4421-8309-fc16e970dca3)
![Screenshot 2023-11-28 at 4 09 27 PM](https://github.com/DataBiosphere/terra-ui/assets/117680805/880c4eae-df2b-4450-b0c6-fdc375a1bf20)

Reply:
If we get a 404 on deletion, I think it's fine to just move on. If the workspace actually does still exist, we will attempt to delete it in again in the future.

Reply:
gotcha